backend/src/routes/v2/model/file/getDownloadFile.ts

```diff
@@ -0,0 +1,56 @@
+import bodyParser from 'body-parser'
+import contentDisposition from 'content-disposition'
+import { Request, Response } from 'express'
+import { z } from 'zod'
+
+import { FileInterface } from '../../../../models/v2/File.js'
+import { downloadFile, getFileById } from '../../../../services/v2/file.js'
+import { BadReq, InternalError } from '../../../../utils/v2/error.js'
+import { parse } from '../../../../utils/validate.js'
+
+export const getDownloadFileSchema = z.object({
+  params: z.object({
+    modelId: z.string({
+      required_error: 'Must specify model id as param',
+    }),
+    fileId: z.string(),
+  }),
+})
+
+interface GetDownloadFileResponse {
+  files: Array<FileInterface>
+}
+
+export const getDownloadFile = [
+  bodyParser.json(),
+  async (req: Request, res: Response<GetDownloadFileResponse>) => {
+    const {
+      params: { fileId },
+    } = parse(req, getDownloadFileSchema)
+
+    const file = await getFileById(req.user, fileId)
+
+    // required to support utf-8 file names
+    res.set('Content-Disposition', contentDisposition(file.name, { type: 'inline' }))
+    res.set('Content-Type', file.mime)
+    res.set('Cache-Control', 'public, max-age=604800, immutable')
+
+    if (req.headers.range) {
```

Comment:
Do this first before logic for response?

Reply:
Range requests will include the same headers (Content-Type, Cache-Control and Content-Disposition) seen above.  The idea behind placing it here is that when we _do_ support it you can just write the implementation in here and not need to change anything else.

I could move the error up (rule: do error checking first), but it feels like a waste for now.